### PR TITLE
CLDR-17544 In en, remove yMd in chinese calendar availableFormats

### PR DESCRIPTION
**Pull request by macchiati**
CLDR-17544

This removes a value from root that isn't really needed, and fixes the example cited in the ticket.

It also adds an option to SearchCLDR to locate these kinds of issues, but doesn't address them all. 
For that I'll clone the ticket.

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Review by pedberg-icu — APPROVED** [review]

**Comment by macchiati** [comment]
Please review

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/SearchCLDR.java [hunk 1]
```diff
@@ -274,14 +301,32 @@ public static void main(String[] args) {
...
-            for (String path : file.fullIterable()) {
-                if (locale.equals("eo") && path.contains("type=\"MK\"")) {
+            final Iterable<String> pathSource =
+                    rootUncovered ? (Iterable<String>) ROOT : file.fullIterable();
+            RelatedPaths relatedPathsWithNonNullValues = new RelatedPaths();
+
+            for (String path : pathSource) {
+                if (path.contains("yMd") && path.contains("chinese")) {
                     int debug = 0;
                 }
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/tool/SearchCLDR.java:311]:
```suggestion
```

you should be able to do these with a conditional breakpoint, anyway

File: tools/cldr-code/src/main/java/org/unicode/cldr/tool/SearchCLDR.java [hunk 2]
```diff
@@ -274,14 +301,32 @@ public static void main(String[] args) {
...
+                if (pathMatcher != null && !pathMatcher.find(path)) {
+                    continue;
+                }
                 String stringValue = file.getStringValue(path);
-                if (stringValue == null) {
+                if (rootUncovered) {
+                    if (stringValue != null) {
+                        // Record any cases where there are non-null, non-inherited values
+                        if (!stringValue.equals("↑↑↑")) {
```
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/tool/SearchCLDR.java:319]:
```suggestion
                        if (!stringValue.equals(CldrUtility.INHERITANCE_MARKER)) {
```

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPaths.java [hunk 3]
```diff
@@ -122,26 +122,31 @@ public void TestPathHeadersAndValues() {
...
             }
             CLDRFile file = testInfo.getCLDRFile(locale, true);
             logln("Testing path headers and values for locale => " + locale);
             final Collection<String> extraPaths = file.getExtraPaths();
+
             for (Iterator<String> it = file.iterator(); it.hasNext(); ) {
                 String path = it.next();
-                if (extraPaths.contains(path)) {
+                if (isExemptLocale && path.equals(exemptPathIfLocale)) {
+                    logKnownIssue("CLDR-17544", "Can't reproduce locally");
```
Comment (srl295) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPaths.java:142]: ^ the message isn't clear as to what's happening.

**Review by srl295 — APPROVED** [review]
LGTM though suggestions for improvement

**Comment by macchiati** [comment]
I'll make a followup PR with the tool improvements, thanks.

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPaths.java [hunk 4]
```diff
@@ -122,26 +122,31 @@ public void TestPathHeadersAndValues() {
...
                     continue;
```
Comment (srl295) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestPaths.java:143]: Post script: This should have been `if(logKnownIssue(…)) continue;` - only continue if logKnownIssue returns true.

**Pull request merged**